### PR TITLE
Make h5py optional

### LINKED ISSUE #924: Make h5py an optional dependancy
h5py requires a compilation step if there are no wheels. As we only use it for legacy file loading we could make it an optional dependency.

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #924, #1204

### CODE REVIEW

File: python/tskit/formats.py [hunk 1]
```diff
@@ -225,6 +224,17 @@ def _load_legacy_hdf5_v3(root, remove_duplicate_positions):
     return tables.tree_sequence()
 
 
+def get_h5py():
+    try:
+        import h5py
+    except ImportError:
+        raise tskit.FileFormatError(
```
Comment (jeromekelleher) [inline python/tskit/formats.py:231]: I think this is still best as an ImportError - we just raise one with a better error message. There's nothing actually wrong with the file format, we're just missing an import.

**Review by jeromekelleher — COMMENTED** [review]
LGTM, one comment.

File: python/tskit/formats.py (hunk 1 above)
Comment (benjeffery) [reply python/tskit/formats.py:231]: Heh, I originally had this as ImportError so happy to change back!

**Comment by benjeffery** [comment]
Fixed up, should be good to go.

**Review by jeromekelleher — APPROVED** [review]
LGTM - merge as you see fit.

**Comment by jeromekelleher** [comment]
Hmm, that circle CI error is a real issue - looks like different versions of numpy somewhere in our CI image.

I vote we skip the test for now, file and issue and look at it again after release?

**Comment by benjeffery** [comment]
@mergifyio rebase

**Pull request merged**